File: src/01-easy/horse-racing-duals/solution.swift
let n = Int(readLine()!)!
var h = (0..<n).map { _ in Int(readLine()!)! }
h.sort()
var m = h[1] - h[0]
for i in 1..<n-1 {
    let d = h[i+1] - h[i]
    if d < m { m = d }
}
print(m)

File: src/01-easy/horse-racing-duals/solution.fs
open System
let n = Console.ReadLine() |> int
let h = [| for _ in 1..n -> Console.ReadLine() |> int |] |> Array.sort
let m = Array.init (n-1) (fun i -> h.[i+1] - h.[i]) |> Array.min
printfn "%d" m
